db/schema_tables.cc

```diff
@@ -2460,9 +2460,11 @@ static index_metadata create_index_from_index_row(const query::result_set_row& r
 
 /*
  * View metadata serialization/deserialization.
+ * If the base schema is not provided, the schema context must have a reference to the database,
+ * and the most up-to-date base schema will be pulled from there.
  */
```

Comment:
It's awkward to have such a function with diverging behavior.

Maybe it's better to force the callers to supply the base schema. But the existing optional leads me to believe the behavior is entrenched.

Reply:
This workaround is for cases where we don't know which table is the base table before deserializing it from the schema_mutations (which we're doing in this function. There are 2 such cases:
* in `create_view_from_table_row` - we get the information for the first time from the system table
* in `get_schema_definition` - we get the schema from another node and we only know the schema version of the view beforehand